cmd/jaeger/internal/extension/jaegerstorage/extension.go

```diff
@@ -137,7 +137,7 @@ func newStorageExt(config *Config, telset component.TelemetrySettings) *storageE
 	return &storageExt{
 		config:           config,
 		telset:           telset,
-		factories:        make(map[string]storage.Factory),
+		factories:        make(map[string]tracestore.Factory),
```

Comment:
@yurishkuro Now that we hold the tracestore.Factory here. The helper methods that we added in the previous PR don't work directly because there's no way to do a runtime cast to Purger/SamplingStoreFactory

Reply:
Any thoughts on how we shoulda address this? One way would be to implement these factories in the v1adapter.

Reply:
yes, we can implement them in the adapter, but conditionally, using this technique: https://github.com/opentracing-contrib/go-stdlib/blob/09b6f447e1a1db7ac0a26870ea06117c677ef188/nethttp/metrics-tracker.go#L32

Reply:
https://github.com/jaegertracing/jaeger/pull/6701